modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/file/FilePageStoreManager.java

```diff
@@ -1093,8 +1095,13 @@ private IgniteCheckedException shutdown(Collection<CacheStoreHolder> holders, bo
      */
     private IgniteCheckedException shutdown(CacheStoreHolder holder, boolean cleanFile,
         @Nullable IgniteCheckedException aggr) {
+
         aggr = shutdown(holder.idxStore, cleanFile, aggr);
 
+        if (cleanFile) {
+            cleanParentDirectories(holder);
+        }
```

Comment:
What's the reason behind deleting a directory **before** you close page store instances for its content? This code looks broken, I don't understand your intentions

Reply:
mistake. fixed